```
/work/tt_metal/api/tt-metalium/logger.hpp:57:13: error: unknown pragma ignored [clang-diagnostic-unknown-pragmas]
   57 | #pragma GCC visibility push(hidden)
      |             ^
/work/tt_metal/api/tt-metalium/logger.hpp:189:13: error: unknown pragma ignored [clang-diagnostic-unknown-pragmas]
  189 | #pragma GCC visibility pop
      |             ^
184066 warnings and 2 errors generated.
```

wow... clang doesn't like gcc!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -190,34 +190,52 @@ target_link_libraries(
 )
 
 add_compile_options(
-    -Werror
-    -Wno-deprecated-declarations
-    -Wdelete-non-virtual-dtor
-    -Wreturn-type
-    -Wswitch
-    -Wuninitialized
+    "$<$<STREQUAL:${CMAKE_SYSTEM_PROCESSOR},x86_64>:-mavx2>"
+    -fno-lto # FIXME: This seems to be here for ttnn; it should go to TTNN, then.
     -fPIC
     -fvisibility-inlines-hidden
-    -fno-lto # FIXME: This seems to be here for ttnn; it should go to TTNN, then.
-    "$<$<STREQUAL:${CMAKE_SYSTEM_PROCESSOR},x86_64>:-mavx2>"
+    -Wall
+    -Werror
+    # The following list is not indicative of what we _desire_ only the status quo to build.
+    -Wno-deprecated-declarations
+    -Wno-bool-operation
+    -Wno-pessimizing-move
+    -Wno-range-loop-construct
+    -Wno-unused-but-set-variable
+    -Wno-unused-function
+    -Wno-unused-variable
     "$<$<CXX_COMPILER_ID:Clang>:-Wno-c++11-narrowing>"
-    "$<$<CXX_COMPILER_ID:Clang>:-Wno-error=local-type-template-args>"
-    "$<$<CXX_COMPILER_ID:Clang>:-Wno-delete-non-abstract-non-virtual-dtor>"
-    "$<$<CXX_COMPILER_ID:Clang>:-Wno-c99-designator>"
-    "$<$<CXX_COMPILER_ID:Clang>:-Wno-shift-op-parentheses>"
-    "$<$<CXX_COMPILER_ID:Clang>:-Wno-non-c-typedef-for-linkage>"
     "$<$<CXX_COMPILER_ID:Clang>:-Wno-deprecated-this-capture>"
     "$<$<CXX_COMPILER_ID:Clang>:-Wno-deprecated-volatile>"
-    "$<$<CXX_COMPILER_ID:Clang>:-Wno-deprecated-builtins>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-for-loop-analysis>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-logical-op-parentheses>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-mismatched-tags>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-missing-braces>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-reorder-ctor>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-self-assign-overloaded>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-self-assign>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-self-move>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-shift-op-parentheses>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-unused-lambda-capture>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-unused-local-typedef>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-unused-private-field>"
+    "$<$<CXX_COMPILER_ID:Clang>:-Wno-unknown-pragmas>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-array-bounds>"
     "$<$<CXX_COMPILER_ID:GNU>:-Wno-deprecated>"
-    "$<$<CXX_COMPILER_ID:GNU>:-Wno-attributes>"
-    "$<$<CXX_COMPILER_ID:GNU>:-Wno-stringop-overread>"
-    "$<$<CXX_COMPILER_ID:GNU>:-Wno-stringop-overflow>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-int-in-bool-context>"
     "$<$<CXX_COMPILER_ID:GNU>:-Wno-maybe-uninitialized>"
-    "$<$<CXX_COMPILER_ID:GNU>:-Wno-missing-requires>"
     "$<$<CXX_COMPILER_ID:GNU>:-Wno-narrowing>"
     "$<$<CXX_COMPILER_ID:GNU>:-Wno-non-template-friend>"
-    "$<$<CXX_COMPILER_ID:GNU>:-Wno-error=non-template-friend>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-parentheses>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-pragmas>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-reorder>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-restrict>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-sign-compare>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-strict-aliasing>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-stringop-overflow>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-stringop-overread>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-unused-local-typedefs>"
+    "$<$<CXX_COMPILER_ID:GNU>:-Wno-volatile>"
     "$<$<BOOL:${ENABLE_ASAN}>:-fsanitize=address>"
     "$<$<BOOL:${ENABLE_MSAN}>:-fsanitize=memory>"
     "$<$<BOOL:${ENABLE_TSAN}>:-fsanitize=thread>"

